contracts/L2ForkArbitrator.sol

```diff
@@ -0,0 +1,189 @@
+    IRealityETH public realitio;
+
+    enum RequestStatus {
+        NONE,
+        QUEUED,
+        FORK_REQUESTED, // Fork is happening
+        FORK_COMPLETED,  // Fork has happened
+        REQUEST_FAILED   // TODO: Check if we need this or if we can just put it back to QUEUED
+    }
+
+    struct ArbitrationRequest {
+        RequestStatus status; 
+	address payable payer;
+        uint256 paid;
+	bytes32 result;
+    } 
+
+    event LogRequestArbitration(
+        bytes32 indexed question_id,
+        uint256 fee_paid,
+        address requester,
+        uint256 remaining
+    );
+
+    mapping(bytes32 => ArbitrationRequest) public arbitrationRequests;
+
+    L2ChainInfo public chainInfo;
+    L1GlobalRouter public router;
+
+    uint256 public disputeFee; // The dispute fee should generally only go down
+    uint256 public chainId;
+
+    constructor(IRealityETH _realitio, L2ChainInfo _chainInfo, L1GlobalRouter _router, uint256 _initialDisputeFee) {
+        realitio = _realitio;
+        chainInfo = _chainInfo; 
+        router = _router;
+        disputeFee = _initialDisputeFee;
+    }
+
+    /// @notice Return the dispute fee for the specified question. 0 indicates that we won't arbitrate it.
+    /// @dev Uses a general default, takes a question id param for other contracts that may want to set it per-question.
+    function getDisputeFee(bytes32) public view returns (uint256) {
+        return disputeFee;
+    }
+
+    /// @notice Request arbitration, freezing the question until we send submitAnswerByArbitrator
+    /// @dev The bounty can be paid only in part, in which case the last person to pay will be considered the payer
+    /// Will trigger an error if the notification fails, eg because the question has already been finalized
+    /// @param question_id The question in question
+    /// @param max_previous If specified, reverts if a bond higher than this was submitted after you sent your transaction.
+    function requestArbitration(
+        bytes32 question_id,
+        uint256 max_previous
+    ) external payable returns (bool) {
+        uint256 arbitration_fee = getDisputeFee(question_id);
+        require(
+            arbitration_fee > 0,
+            "fee must be positive"
+        );
+
+        require(arbitrationRequests[question_id].status == RequestStatus.NONE, "Already requested");
+
+        arbitrationRequests[question_id] = ArbitrationRequest(RequestStatus.QUEUED, payable(msg.sender), msg.value, bytes32(0));
+
+	realitio.notifyOfArbitrationRequest(
+	    question_id,
+	    msg.sender,
+	    max_previous
+	);
+	emit LogRequestArbitration(question_id, msg.value, msg.sender, 0);
+
+        if (!isForkInProgress) {
+           requestActivateFork(question_id); 
+        }
+
+	return true;
+    }
+
+    // Request a fork via the bridge
+    // NB This may fail if someone else is already forking, and we'll have to retry after the fork.
+    function requestActivateFork(
+        bytes32 question_id
+    ) public {
```

Comment:
this one must be internal or charge the tokens

Reply:
It shouldn't be callable unless the tokens are already paid because of the status check, but let's check this again once we've put the tokens movement and the failure logic in.